### PR TITLE
CORE-11273 Add missing Javadocs to crypto APIs

### PR DESCRIPTION
**Pull request by kyriathar**
- adds missing Javadocs to crypto APIs
- minor amendments to existing crypto APIs Javadocs

Please note that his PR **needs to also be reviewed by tech writers team**, once devs approve content.

### CODE REVIEW

**Review by driessamyn — DISMISSED** [review]
LGTM

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java [hunk 1]
```diff
@@ -13,10 +12,11 @@ public final class CompositeKeyNodeAndWeight {
...
 
     /**
-     * Construct a key.
+     * Creates a new {@code CompositeKeyNodeAndWeight} for
+     * the specified key and weight for the key.
      *
-     * @param node   A public key.
-     * @param weight The weight for that key, must be greater than zero.
+     * @param node   A public key
+     * @param weight The weight for that key, must be greater than zero
```
Comment (dickon) [inline crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]: We should describe what the weight does

Comment (dickon) [reply crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]: Something like

@param weight The weight for that key, must be greater than zero. The weight controls the contribution a key makes to reaching the threshold that specifies whether the composite key is regarded as fulfilled.

File: crypto/src/main/java/net/corda/v5/crypto/CordaOID.java [hunk 2]
```diff
@@ -7,9 +7,7 @@
  * If an OID is incorrectly assigned, it should be marked deprecated and NEVER be reused again.
  */
 public final class CordaOID {
-    private CordaOID() {
-        // this class is never constructed; it exists for the static strings only
```
Comment (dickon) [inline crypto/src/main/java/net/corda/v5/crypto/CordaOID.java:11]: I don't agree with removing this, I know we talked about it before and people felt it was a standard pattern, but it is better to be explicit.

However I won't block the PR for it,  it's a pretty minor matter.

**Review by dickon — COMMENTED** [review]
We should describe what the weight does.

File: crypto/src/main/java/net/corda/v5/crypto/CordaOID.java (hunk 2 above)
Comment (kyriathar) [reply crypto/src/main/java/net/corda/v5/crypto/CordaOID.java:11]: Yes good point, it's just that's how we have it elsewhere in the API code base in similar cases (i.e. for not constructable classes) so I believe we should be aligned, either remove these or add it everywhere.

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java (hunk 1 above)
Comment (kyriathar) [reply crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]: Agreed, perhaps better address this under a different JIRA?

**Review by dickon — DISMISSED** [review]
LGTM. We should describe the weight in compound keys somewhere, and maybe provide examples, which could be done in CORE-11273 or a separate ticket if @kyriathar prefers.

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java (hunk 1 above)
Comment (dickon) [reply crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]: Ok

**Comment by dickon** [comment]
I think this should target 5.0 and get an exception reviewed; it is currently targetting 5.1.

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java (hunk 1 above)
Comment (kyriathar) [reply crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]: Raised [JIRA](https://r3-cev.atlassian.net/browse/CORE-13722) for it.

File: crypto/src/main/java/net/corda/v5/crypto/merkle/MerkleProof.java [hunk 3]
```diff
@@ -20,17 +20,25 @@
...
     int getTreeSize();
 
+    /**
+     * Returns disclosed plain data of a {@link MerkleTree} which are to be verified that
+     * they are indeed part of a {@code MerkleTree}.
+     */
     List<IndexedMerkleLeaf> getLeaves();
 
+    /**
+     * Returns the hashed data supplementing the plain data.
```
Comment (mnesbit) [inline crypto/src/main/java/net/corda/v5/crypto/merkle/MerkleProof.java]: Returns the hashed intermediate node data supplementing the plain data.

Comment (kyriathar) [reply crypto/src/main/java/net/corda/v5/crypto/merkle/MerkleProof.java]: Suggestion applied

**Review by mnesbit — DISMISSED** [review]
OK

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java [hunk 4]
```diff
@@ -13,10 +12,11 @@ public final class CompositeKeyNodeAndWeight {
     private final int weight;
 
     /**
-     * Construct a key.
+     * Creates a new {@code CompositeKeyNodeAndWeight} for
+     * the specified key and weight for the key.
      *
-     * @param node   A public key.
-     * @param weight The weight for that key, must be greater than zero.
+     * @param node   A public key
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]:
```suggestion
     * @param node   A public key.
```

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java (hunk 1 above)
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]:
```suggestion
     * @param weight The weight for that key, must be greater than zero.
```

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java [hunk 5]
```diff
@@ -25,15 +25,27 @@ public CompositeKeyNodeAndWeight(@NotNull PublicKey node, int weight) {
         this.weight = weight;
     }
 
+    /**
+     * Creates a new {@code CompositeKeyNodeAndWeight} for
+     * the specified key, defaulting the key's weight to 1.
+     *
+     * @param node  A public key
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]:
```suggestion
     * @param node  A public key.
```

File: crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java [hunk 6]
```diff
@@ -86,12 +87,12 @@ public static boolean isKeyFulfilledBy(@NotNull PublicKey key, @NotNull Set<Publ
...
      * is the same as calling as the variant that takes an iterable set of other keys with <code>otherKey<code>
      * as a single element iterable. 
-     * 
+     * <p>
      * Since we do not define composite keys as acceptable on the second argument of this function, this relation
      * is not reflexive, not symmetric and not transitive. 
      *
-     * @param key The key with the requirements.
-     * @param otherKey The key to check whether requirements are fulfilled.
+     * @param key The key to be checked whether it is being fulfilled by {@code otherKey}
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java]:
```suggestion
     * @param key The key to be checked whether it is being fulfilled by {@code otherKey}.
```

File: crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java [hunk 7]
```diff
@@ -86,12 +87,12 @@ public static boolean isKeyFulfilledBy(@NotNull PublicKey key, @NotNull Set<Publ
...
+     * @param otherKey The key against which the {@code key} is being checked for fulfillment
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java]:
```suggestion
     * @param otherKey The key against which the {@code key} is being checked for fulfilment.
```

File: crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java [hunk 8]
```diff
@@ -71,8 +72,8 @@ public static boolean isKeyInSet(@NotNull PublicKey key, @NotNull Set<PublicKey>
      * check fulfilment against a set of keys, without having to handle simple and composite keys separately (that is, this is
      * polymorphic).
      * 
-     * @param key  The key with the requirements.
-     * @param otherKeys The key to check whether requirements are fulfilled.
+     * @param key  The key to be checked whether it is being fulfilled by {@code otherKeys}
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java]:
```suggestion
     * @param key  The key to be checked whether it is being fulfilled by {@code otherKeys}.
```

File: crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java [hunk 9]
```diff
@@ -71,8 +72,8 @@ public static boolean isKeyInSet(@NotNull PublicKey key, @NotNull Set<PublicKey>
...
+     * @param otherKeys The keys against which the {@code key} is being checked for fulfillment
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java]:
```suggestion
     * @param otherKeys The keys against which the {@code key} is being checked for fulfilment.
```

File: crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java [hunk 10]
```diff
@@ -38,9 +39,9 @@ private KeyUtils() {
      * Then we would find that <code>isKeyInSet(C, X)</code> is true, but X would not fulfill C since C is fulfilled by
      * X and Y together but not X on its. However, <code>isKeyInSet(C, Z)</code> is true, and Z fulfills C by itself.
      * 
-     * @param key       The key being checked for.
-     * @param otherKeys An {@link Iterable} sequence of {@link PublicKey}.
-     * @return True if <code>key</code> is in otherKeys.
+     * @param key       The key being looked for
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java]:
```suggestion
     * @param key       The key being looked for.
```

File: crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java [hunk 11]
```diff
@@ -38,9 +39,9 @@ private KeyUtils() {
...
+     * @param otherKeys The keys to look the {@code key} in
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java]:
```suggestion
     * @param otherKeys The keys searched for the {@code key}.
```

File: crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java [hunk 12]
```diff
@@ -38,9 +39,9 @@ private KeyUtils() {
...
+     * @return True if <code>key</code> is in otherKeys
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java]:
```suggestion
     * @return True if <code>key</code> is in otherKeys.
```

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java [hunk 13]
```diff
@@ -25,15 +25,27 @@ public CompositeKeyNodeAndWeight(@NotNull PublicKey node, int weight) {
...
+     */
     public CompositeKeyNodeAndWeight(@NotNull PublicKey node) {
         this(node, 1);
     }
 
+    /**
+     * @return the key of the {@code CompositeKeyNodeAndWeight}
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]:
```suggestion
     * @return The key of the {@code CompositeKeyNodeAndWeight}.
```

File: crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java [hunk 14]
```diff
@@ -25,15 +25,27 @@ public CompositeKeyNodeAndWeight(@NotNull PublicKey node, int weight) {
...
+     */
     @NotNull
     public PublicKey getNode() {
         return this.node;
     }
 
+    /**
+     * @return the weight of the key
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java]:
```suggestion
     * @return The weight of the key.
```

File: crypto/src/main/java/net/corda/v5/crypto/DigitalSignature.java [hunk 15]
```diff
@@ -11,6 +11,9 @@
 @CordaSerializable
 public interface DigitalSignature {
 
+    /**
+     * @return the digital signature bytes
```
Comment (amiegrace12) [inline crypto/src/main/java/net/corda/v5/crypto/DigitalSignature.java]:
```suggestion
     * @return The digital signature bytes.
```

**Review by amiegrace12 — APPROVED** [review]

**Review by driessamyn — APPROVED** [review]
Approved (doc changes only, this change does not affect code).

**Pull request merged**